bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/SonyAuth.java

```diff
@@ -0,0 +1,393 @@
+        }
+
+        final IrccClient irccClient = getIrccClient == null ? null : getIrccClient.getClient();
+        return irccClient == null ? null : NetUtil.getSonyUrl(irccClient.getBaseUrl(), ScalarWebService.ACCESSCONTROL);
+    }
+
+    /**
+     * Request access by initiating the registration or doing the activation if on the second step
+     *
+     * @param transport a non-null transport to use
+     * @param accessCode the access code (null for initial setup)
+     * @return the http response
+     */
+    public AccessResult requestAccess(final SonyHttpTransport transport, @Nullable final String accessCode) {
+        Objects.requireNonNull(transport, "transport cannot be null");
+
+        logger.debug("Requesting access: {}", StringUtils.defaultIfEmpty(accessCode, "(request)"));
+
+        if (accessCode != null) {
+            transport.setOption(new TransportOptionHeader(NetUtil.createAccessCodeHeader(accessCode)));
+        }
+        transport.setOption(new TransportOptionHeader(getDeviceIdHeaderName(), NetUtil.getDeviceId()));
+
+        final ScalarWebResult result = scalarActRegister(transport, accessCode);
+        final HttpResponse httpResponse = result.getHttpResponse();
+
+        final String registrationUrl = getRegistrationUrl();
+        if (httpResponse.getHttpCode() == HttpStatus.UNAUTHORIZED_401) {
+            if (registrationUrl == null || StringUtils.isEmpty(registrationUrl)) {
+                return accessCode == null ? AccessResult.PENDING : AccessResult.NOTACCEPTED;
+            }
+        }
+
+        if (result.getDeviceErrorCode() == ScalarWebError.NOTIMPLEMENTED
+                || (result.getDeviceErrorCode() == ScalarWebError.HTTPERROR
+                        && httpResponse.getHttpCode() == HttpStatus.SERVICE_UNAVAILABLE_503)
+                || httpResponse.getHttpCode() == HttpStatus.UNAUTHORIZED_401
+                || httpResponse.getHttpCode() == HttpStatus.FORBIDDEN_403) {
+            if (registrationUrl != null && StringUtils.isNotEmpty(registrationUrl)) {
+                final HttpResponse irccResponse = irccRegister(transport, accessCode);
+                if (irccResponse.getHttpCode() == HttpStatus.OK_200) {
+                    return AccessResult.OK;
+                } else if (irccResponse.getHttpCode() == HttpStatus.UNAUTHORIZED_401) {
+                    return AccessResult.PENDING;
+                } else {
+                    return new AccessResult(irccResponse);
+                }
+            }
+        }
+
+        if (result.getDeviceErrorCode() == ScalarWebError.DISPLAYISOFF) {
+            return AccessResult.DISPLAYOFF;
+        }
+
+        if (httpResponse.getHttpCode() == HttpStatus.SERVICE_UNAVAILABLE_503) {
+            return AccessResult.HOMEMENU;
+        }
+
+        if (httpResponse.getHttpCode() == HttpStatus.OK_200
+                || result.getDeviceErrorCode() == ScalarWebError.ILLEGALARGUMENT) {
+            return AccessResult.OK;
```

Comment:
Illegal argument is ok?

Reply:
Yes - but this is a section of code I've rewritten based on the new device and I've documented why